editor/editor_log.cpp

```diff
@@ -267,21 +267,59 @@ void EditorLog::_undo_redo_cbk(void *p_self, const String &p_name) {
 void EditorLog::_rebuild_log() {
 	log->clear();
 
-	for (int msg_idx = 0; msg_idx < messages.size(); msg_idx++) {
+	int line_limit = int(EDITOR_GET("run/output/max_lines"));
```

Comment:
It might be worth caching this (and the other `EDITOR_GET()` call below) to a member variable, so we don't have to query project settings several times per frame if `print()` is being used every frame in the project.

Reply:
Fixed. Added callback for Editor Setting changes and updates the `line_limit` if needed. `EDITOR_GET` should only be called then and at startup, not during usual operation.